libr/search/search.c

```diff
@@ -427,7 +406,9 @@ R_API RList *r_search_find(RSearch *s, ut64 addr, const ut8 *buf, int len) {
 
 /* --- keywords --- */
 R_API int r_search_kw_add(RSearch *s, RSearchKeyword *kw) {
-	if (!kw) return false;
```

Comment:
The call sites should do the check.

```c
	case 'a': if (input[1]) {
...........
				r_search_kw_add (core->search,
					r_search_keyword_new_hexmask (kwd, NULL)); ///// it should check if new_hexmask returns NULL
				r_search_begin (core->search);
				free (kwd);
				dosearch = true;
```

The call site are closer to the user and have more knowledge about the interactive context. It can give useful warnings/errors and it should have do that. Then such check in the search.c side will be redundant. If the library is full of such `if (!xxx) return;` checks, there will not be much integrity guarantee and you'll tend to distrust everything.